nevergrad/functions/photonics/photonics.py

```diff
@@ -257,42 +255,51 @@ def cascade2(A: np.ndarray, B: np.ndarray) -> np.ndarray:
 
 
 def solar(lam: np.ndarray) -> np.ndarray:
+    # saved with np.save(filename, e) and dumped in this folder
     a = np.load(Path(__file__).with_name("wavelength_solar.npy"))
-    e = np.load(Path(__file__).with_name("epsilon_solar.npy"))  # saved with np.save(filename, e) and dumped in this folder
+    e = np.load(Path(__file__).with_name("epsilon_solar.npy"))
     jsc = np.interp(lam, a, e)
     return jsc  # type: ignore
 
 
-def absorption(lam, Epsilon, Mu, Type, hauteur, pol, theta):
-    if pol == 0:
-        f = Mu
-    else:
-        f = Epsilon
+def absorption(
+    lam: float,
+    epsilon: np.ndarray,
+    mu: np.ndarray,
+    type_: np.ndarray,
+    hauteur: np.ndarray,
+    pol: float,
+    theta: int,
+) -> np.ndarray:
```

Comment:
@Ellawin @AnMoreau I've added type hints, and slightly renamed the variables (renaming capitalized letters) since this is more "standard" this way.

Reply:
pol is an int, and theta a float